### PR TITLE
optimisation: stepper: make `dda_isteps_t` an array

### PR DESCRIPTION
**Pull request by gudnimg**
This way we can loop through it with a for-loop
and save some code size

### CODE REVIEW

File: Firmware/stepper.cpp [hunk 1]
```diff
@@ -393,18 +389,20 @@ FORCE_INLINE void stepper_next_block()
     }
 
     if (current_block->flag & BLOCK_FLAG_DDA_LOWRES) {
-      counter_x.lo = -(current_block->step_event_count.lo >> 1);
-      counter_y.lo = counter_x.lo;
-      counter_z.lo = counter_x.lo;
-      counter_e.lo = counter_x.lo;
+      const int16_t value = -(current_block->step_event_count.lo >> 1);
+      for (uint8_t axis = 0; axis < NUM_AXIS; axis++)
```
Comment (DRracer) [inline Firmware/stepper.cpp:393]: you can even iterate from the back which saves 2-4 bytes more
```
for (uint8_t axis = NUM_AXIS-1; axis >=0 ; --axis){
    counter[axis].lo = value;
}
```
Caches and memory ordering is not an issue on the AVR IMHO

**Review by DRracer — APPROVED** [review]
:thinking: most of the savings in this PR come from setting `counter_*` variables in loops. The rest is just a mechanical rewrite to match the proposed "array" principle.

I'm surprised that the compiler understood the intent and actually saved something while not breaking the rest of the code.

File: Firmware/stepper.cpp (hunk 1 above)
Comment (Ro3Deee) [reply Firmware/stepper.cpp:393]: what about this, using postdecrementation :
`for (uint8_t axis = NUM_AXIS; axis-- ; ){`
`    counter[axis].lo = value;`
`}` ?

Comment (3d-gussner) [reply Firmware/stepper.cpp:393]: @petrubecheru Thanks but compiling with `PF-build.sh` the usage of Flash and RAM are the same before and after.

**Pull request merged**